### ISSUE TITLE
Package version issue

### ISSUE THREAD

**Issue by patricsgamma** [payloadcms/payload#3929]
### Link to reproduction

-

### Describe the Bug

Updating to the latest version of Payload 2.x breaks [plugin-sentry](https://github.com/payloadcms/plugin-sentry)

### Working state ✅ 
We are currently at `"payload": "^2.0.6"` including following packages. 
  ```
  "@payloadcms/bundler-webpack": "^1.0.4",
    "@payloadcms/db-mongodb": "^1.0.4",
    "@payloadcms/plugin-cloud-storage": "^1.1.1",
    "@payloadcms/plugin-sentry": "^0.0.4",
    "@payloadcms/richtext-lexical": "^0.1.11",
````

### Broken state ❌ 
Upgrading to `"payload": "^2.0.13"` with latest packages
   ```
 "@payloadcms/bundler-webpack": "^1.0.4",
    "@payloadcms/db-mongodb": "^1.0.4",
    "@payloadcms/plugin-cloud-storage": "^1.1.1",
    "@payloadcms/plugin-sentry": "^0.0.5",
    "@payloadcms/richtext-lexical": "^0.1.15",
```
results in following error 🐛 

### Error Description
```
yarn generate:types
yarn run v1.22.19
$ cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:types
node:internal/modules/cjs/loader:1072
  const err = new Error(message);
              ^

Error: Cannot find module '@sentry/react'
Require stack:
- /Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/mocks/component.js
- /Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/plugin.js
- /Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/index.js
- /Users/x/Develop/y/Portal.Cms/src/payload.config.ts
- /Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/config/load.js
- /Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/bin/generateGraphQLSchema.js
- /Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/bin/index.js
- /Users/x/Develop/y/Portal.Cms/Portal.Cms/node_modules/payload/bin.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1072:15)
    at Module._load (node:internal/modules/cjs/loader:925:27)
    at Module.require (node:internal/modules/cjs/loader:1139:19)
    at require (node:internal/modules/helpers:121:18)
    at Object.<anonymous> (/Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/mocks/component.js:67:27)
    at Module._compile (node:internal/modules/cjs/loader:1257:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1311:10)
    at Object.newLoader [as .js] (/Users/x/Develop/y/Portal.Cms/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1115:32)
    at Module._load (node:internal/modules/cjs/loader:962:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/mocks/component.js',
    '/Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/plugin.js',
    '/Users/x/Develop/y/Portal.Cms/node_modules/@payloadcms/plugin-sentry/dist/index.js',
    '/Users/x/Develop/y/Portal.Cms/src/payload.config.ts',
    '/Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/config/load.js',
    '/Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/bin/generateGraphQLSchema.js',
    '/Users/x/Develop/y/Portal.Cms/node_modules/payload/dist/bin/index.js',
    '/Users/x/Develop/y/Portal.Cms/node_modules/payload/bin.js'
  ]
}

Node.js v20.3.1
```

Adding the `yarn add @sentry/react`  manually lets me overcome the package issue.
Which throws me into an other error.

```
yarn generate:types
yarn run v1.22.19
$ cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:types
/Users/x/Develop/y/Portal.Cms/src/collections/eduXml/eduxml-simulations.ts:88
            relationTo: _.DisciplineAlignments.slug,
                                               ^

TypeError: Cannot read properties of undefined (reading 'slug')
    at Object.<anonymous> (/Users/x/Develop/y/Portal.Cms/src/collections/eduXml/eduxml-simulations.ts:88:48)
    at Module._compile (node:internal/modules/cjs/loader:1257:14)
    at Module._compile (/Users/x/Develop/y/Portal.Cms/node_modules/pirates/lib/index.js:117:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1311:10)
    at Object.newLoader [as .ts] /Users/x/Develop/y/Portal.Cms/node_modules/pirates/lib/index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:1115:32)
    at Module._load (node:internal/modules/cjs/loader:962:12)
    at Module.require (node:internal/modules/cjs/loader:1139:19)
    at require (node:internal/modules/helpers:121:18)
    at Object.<anonymous> (/Users/x/Develop/y/Portal.Cms/src/collections/index.ts:8:14)

Node.js v20.3.1
error Command failed with exit code 1.
```

The two solutions that work are
- To upgrade payload and lexical and remove `@payloadcms/plugin-sentry`
- Stay at payload: `2.0.6` and lexical: `0.1.11` and have `@payloadcms/plugin-sentry`

Not 100% sure if you want this reported here or at the [plugin-sentry Repo](https://github.com/payloadcms/plugin-sentry).
But looks like its a X package version problem.

### To Reproduce

See description

### Payload Version

2.0.13

### Adapters and Plugins

-

**Comment by denolfe** [comment]
Thank you for the report @patricsgamma . This will be resolved once #3671 is merged and released. Will bump this up in priority.

**Comment by denolfe** [comment]
@patricsgamma Released a beta version `@payloadcms/plugin-sentry@0.0.6-beta.0`. Can you give this a try and see if it works for you?
